### DISCUSSION TITLE
CLion

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1203] [Team Posts]
Howdy, more spamming.  ;)

I was recently reminded about CLion. It is arguably the best IDE there is for C++, and there is a special license option for open source projects so it is free to use.
Back in 2015-2016 we had applied for this version but for whatever reason it was not renewed.

Thought I would just chime in that if anyone wants to try it again we can reapply.
Made in java, so a bit of a resource hog, and not open source.. but still an extremely competent tool.

**Comment by kybos** [comment]
Yes, the IDEs from JetBrains are very very good.
Interesting, I did not know there was a way to get a free license for CLion, I knew only about the Python and Java ones.
I would be definitely interested to try it out. Not sure if it would replace qtCreator in the long term, which certainly has its benefits for Qt development, but sometimes it's driving me crazy, so who knows?

**Comment by terminator356** [comment]
Bah. You can replace my KDevelop when you "pry it from my cold dead hands" ;-)

**Comment by spamatica** [comment]
> Bah. You can replace my KDevelop when you "pry it from my cold dead hands" ;-)

I had a feeling you would say something like that :D
Kind of the same way I feel about MusE sometimes. ;)
Are you contributing to kdevelop btw?

In any case, after I sent the question yesterday I figured there was nothing to lose by sending the application so I already took care of that. The reply said it could take up to two weeks, will keep you posted.

**Comment by spamatica** [comment]
I got the licenses!
```
Robert SD47E96916
Kybos  Y2SQP3PIDL
Tim    O55DC9M1EL
```
They aren't actually keyed to a specific person I just assigned them for convenience, one to Tim too, in case he finds himself in the afterlife and needs to use CLion :D

**Comment by spamatica** [comment]
Here is the document they sent. Haven't tried activating, maybe it is needed. Possibly it needs to be activated within one week too.
https://drive.google.com/file/d/164AZSbOsncfbksCm3WyNrH5aA-HagtRN/view?usp=sharing

**Comment by spamatica** [comment]
I went through the activation steps, it was not completely clear which steps need to be done.
I think you need to register an account at jet brains and then visit this link...maybe?!
https://account.jetbrains.com/a/hdcve8r5

**Comment by kybos** [comment]
Yes, a little confusing in the beginning, but after I created my account and then clicked the link above, it worked, I could download my license key and enter it in clion.
Thanks!
